src/Mailgun/Model/Message/ShowResponse.php

```diff
@@ -142,7 +142,8 @@ public static function create(array $data)
             $response->setMessageUrl($data['message-url']);
         }
         if (isset($data['message-headers'])) {
-            $response->setMessageHeaders($data['message-headers']);
+            $headers = is_string($data['message-headers']) ? json_decode($data['message-headers']) : $data['message-headers'];
```

Comment:
As I said, I will provide a test...when I get around to it. In the meantime, as a work around, I am already doing the following:
`$data= <object I got from the POST>`
$data['message-headers'] = json_decode($data['message-headers']);
ShowResponse::create($data);`

And that's working great. So it's working for me...